hail/build.mill

```diff
@@ -171,7 +123,50 @@ trait HailScalaModule extends SbtModule with ScalafmtModule with ScalafixModule
   }
 }
 
-object main extends RootModule with HailScalaModule { outer =>
+object `package` extends RootModule with HailScalaModule { outer =>
+
+  object env extends Module {
+    def scalaVersion: T[String] = Task.Input {
+      val v = T.ctx().env.getOrElse("SCALA_VERSION", "2.12.15")
+      if (!v.startsWith("2.12"))
+        Result.Failure("Hail currently supports only Scala 2.12")
+      else
+        v
+    }
+
+    def javaVersion: T[String] = Task.Input {
+      System.getProperty("java.version")
+    }
+
+    def sparkVersion: T[String] = Task.Input {
+      Result.Success(T.ctx().env.getOrElse("SPARK_VERSION", "3.5.0"))
+    }
+
+    def debugMode: T[Boolean] = Task.Input {
+      val isDebug = !T.ctx().env.contains("HAIL_RELEASE_MODE")
+      T.log.info(s"Building in ${if (isDebug) "debug" else "release"} mode")
+      isDebug
+    }
+
+    def debugOrRelease: Task[String] = Task.Anon {
+      if (debugMode()) "debug" else "release"
+    }
```

Comment:
I think this should be a little more shouty. For example, when producing a debug build, something like the following could be written to the console:
```
Building in DEBUG mode: Optimisation is OFF, JVM Memory Instrumentation is ON
```